commercetools-sdk-base/src/main/java/io/sphere/sdk/client/CorrelationIdGenerator.java

```diff
@@ -0,0 +1,31 @@
+package io.sphere.sdk.client;
+
+/**
+ * Generates correlation ids for each request.
+ *
+ * The format of the correlation id is defined as:
+ * {@code <projectKey> / <UUID> (/ customId)*}
+ */
+public interface CorrelationIdGenerator {
```

Comment:
Also it is not necessary to expose this interface, I don't see it as use case that clients want to override this. Even if they want to override it, they could use a SphereClientDecorator.